src/CMakeLists.txt

```diff
@@ -112,29 +185,103 @@ include_directories(${GLFW_INCLUDE_DIRS})
 include_directories(${JSONCPP_INCLUDE_DIRS})
 link_directories(${GLFW_LIBRARY_DIRS})
 link_directories(${JSONCPP_LIBRARY_DIRS})
+include_directories(${PROJECT_SOURCE_DIR})
 
 # Open3D libraries
 add_subdirectory(Core)
 add_subdirectory(Tools)
 add_subdirectory(IO)
 add_subdirectory(Visualization)
 
-message(STATUS "Open3D_BUILD_TESTS=${Open3D_BUILD_TESTS}")
-if(Open3D_BUILD_TESTS)
+add_library(${CMAKE_PROJECT_NAME} $<TARGET_OBJECTS:Core> $<TARGET_OBJECTS:IO> $<TARGET_OBJECTS:Visualization>)
+
+# headless rendering
+if (ENABLE_HEADLESS_RENDERING)
+    find_package(OSMesa REQUIRED)
+    include_directories(${OSMESA_INCLUDE_DIR})
+    target_link_libraries(${CMAKE_PROJECT_NAME} ${OSMESA_LIBRARY})
+else ()
+    find_package(OpenGL REQUIRED)
+    target_link_libraries(${CMAKE_PROJECT_NAME} ${OPENGL_LIBRARIES})
+endif ()
+
+target_link_libraries(${CMAKE_PROJECT_NAME}
+                      ${GLFW_LIBRARIES}
+                      ${GLEW_LIBRARIES}
+                      ${JPEG_LIBRARIES}
+                      ${JSONCPP_LIBRARIES}
+                      ${OMP_LIBRARIES}
+                      ${PNG_LIBRARIES}
+                      tinyfiledialogs)
```

Comment:
Missing {}?

Reply:
There's no tinyfiledialogs_LIBRARIES variable to use here, tinyfiledialogs always gets built from source. I guess I can manually add it - would you prefer that?

Reply:
Yes, if it can be addressed quickly. This will make cmake file more consistent.

Reply:
done.